tests/test_main.py

```diff
@@ -1374,3 +1376,47 @@ class M(BaseModel):
         a: int
 
     get_type_hints(M.__config__)
+
+
+def foo(arg1, arg2):
+    return arg1, arg2
+
+
+@pydantic_dataclass
```

Comment:
where possible please define models and pydantic dataclasses inside tests. Most of the logic we're testing here occurs then the type is created.